backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/notification/impl/UserNotificationsConfigRepository.java

```diff
@@ -54,32 +58,78 @@
 
 public class UserNotificationsConfigRepository implements IUserNotificationsConfigRepository
 {
-	private final CCache<UserId, UserNotificationsConfig> userNotificationsConfigsByUserId = CCache.<UserId, UserNotificationsConfig> builder()
+	private final CCache<UserId, UserNotificationsConfig> userNotificationsConfigsByUserId = CCache.<UserId, UserNotificationsConfig>builder()
 			.tableName(I_AD_User_NotificationGroup.Table_Name)
 			.initialCapacity(100)
 			.cacheMapType(CacheMapType.LRU)
 			.additionalTableNameToResetFor(org.compiere.model.I_AD_User.Table_Name)
 			.additionalTableNameToResetFor(I_AD_NotificationGroup.Table_Name)
 			.build();
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final INotificationGroupNameRepository notificationGroupNamesRepo = Services.get(INotificationGroupNameRepository.class);
+	private final IUserDAO userDAO = Services.get(IUserDAO.class);
+	private final IBPartnerOrgBL bPartnerOrgBL = Services.get(IBPartnerOrgBL.class);
 
 	@Override
 	public UserNotificationsConfig getByUserId(final UserId adUserId)
 	{
 		return userNotificationsConfigsByUserId.getOrLoad(adUserId, this::retrieveUserNotificationsConfig);
 	}
 
+	@Override
+	public List<UserId> getByNotificationGroupAndOrgId(final @NonNull NotificationGroupName notificationGroupName, @NonNull final OrgId orgId)
+	{
+		final int notificationGroupId = notificationGroupNamesRepo.getNotificationGroupId(notificationGroupName);
+
+		final boolean restrictToOrgBPUsers = notificationGroupNamesRepo.isNotifyOrgBpUsersOnly(notificationGroupName);
+
+		final List<UserId> userIdsToBeNotified = queryBL
+				.createQueryBuilderOutOfTrx(I_AD_User_NotificationGroup.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_AD_User_NotificationGroup.COLUMN_AD_NotificationGroup_ID, notificationGroupId)
+				.addEqualsFilter(I_AD_User_NotificationGroup.COLUMN_AD_Org_ID, orgId)
+				.create()
+				.listDistinct(I_AD_User_NotificationGroup.COLUMNNAME_AD_User_ID, UserId.class);
+		if (restrictToOrgBPUsers)
+		{
+			userIdsToBeNotified.retainAll(bPartnerOrgBL.retrieveOrgUsers(orgId));
+		}
+		return userIdsToBeNotified;
+	}
+
+	@Override
+	public void create(@NonNull final UserNotificationGroupCreateRequest request)
+	{
+		final I_AD_User_NotificationGroup userNotificationGroup = InterfaceWrapperHelper.newInstance(I_AD_User_NotificationGroup.class);
```

Comment:
i assume we have a unique index which fails in case we insert duplicates, right?

Reply:
yes
```
CREATE UNIQUE INDEX ad_user_notificationgroup_uq
    ON ad_user_notificationgroup (ad_user_id, ad_notificationgroup_id)
;
```